README.rst

```diff
@@ -178,62 +178,86 @@ GetComponentImages example:
     image = Image.open('/usr/src/tesseract/testing/phototest.tif')
     with PyTessBaseAPI() as api:
         api.SetImage(image)
-        boxes = api.GetComponentImages(RIL.TEXTLINE, True)
-        print('Found {} textline image components.'.format(len(boxes)))
-        for i, (im, box, _, _) in enumerate(boxes):
-            # im is a PIL image object
-            # box is a dict with x, y, w and h keys
-            api.SetRectangle(box['x'], box['y'], box['w'], box['h'])
-            ocrResult = api.GetUTF8Text()
-            conf = api.MeanTextConf()
-            print(u"Box[{0}]: x={x}, y={y}, w={w}, h={h}, "
-                  "confidence: {1}, text: {2}".format(i, conf, ocrResult, **box))
+        results = api.GetComponentImages(RIL.TEXTLINE, True)
+        print('Found {} textline image components.'.format(len(results)))
+        for i, (img, box, _, _) in enumerate(results):
+            # img is a PIL image object of the binarized line
+            # box is a dict with x, y, w and h keys (w.r.t. full image)
+            img.save('phototest_line{0}_{x}x{y}+{w}x{h}.png'.format(i, **box), format='PNG')
 
-Orientation and script detection (OSD):
-```````````````````````````````````````
+GetIterator example:
+````````````````````
 
 .. code:: python
 
     from PIL import Image
-    from tesserocr import PyTessBaseAPI, PSM
+    from tesserocr import PyTessBaseAPI, RIL
 
-    with PyTessBaseAPI(psm=PSM.AUTO_OSD) as api:
-        image = Image.open("/usr/src/tesseract/testing/eurotext.tif")
+    image = Image.open('/usr/src/tesseract/testing/phototest.tif')
+    with PyTessBaseAPI() as api:
         api.SetImage(image)
         api.Recognize()
+        
+        it = api.GetIterator()
+        for line in iterate_level(it, RIL.TEXTLINE):
+            text = line.GetUTF8Text(RIL.TEXTLINE)
+            conf = line.Confidence(RIL.TEXTLINE)
+            bbox = line.BoundingBox(RIL.TEXTLINE)
+            bbox = {'x': int(bbox[0]),
+                    'y': int(bbox[1]),
+                    'w': int(bbox[2])-int(bbox[0]),
+                    'h': int(bbox[3])-int(bbox[1])}
+            print(u"Box[{0}]: x={x}, y={y}, w={w}, h={h}, "
+                  "confidence: {1}, text: {2}".format(i, conf, text, **bbox))
 
-        it = api.AnalyseLayout()
-        orientation, direction, order, deskew_angle = it.Orientation()
-        print("Orientation: {:d}".format(orientation))
-        print("WritingDirection: {:d}".format(direction))
-        print("TextlineOrder: {:d}".format(order))
-        print("Deskew angle: {:.4f}".format(deskew_angle))
-
-or more simply with ``OSD_ONLY`` page segmentation mode:
+Layout analysis with orientation and deskewing:
```

Comment:
It's important to understand that there are two distinct mechanisms providing orientation detection: the normal page layout analysis (which you can use with any model, including LSTMs) and the dedicated `osd` model (which is legacy-only). It was documented the other way round.